Index: atlctrls.h
===================================================================
--- atlctrls.h	(revision 80)
+++ atlctrls.h	(working copy)
@@ -9,6 +9,9 @@
 // the terms of this license. You must not remove this notice, or
 // any other, from this software.
 
+// Hydra's Changes
+// added CToolBarCtrlT::SetImageList(HIMAGELIST hImageList, int nImageList)
+
 #ifndef __ATLCTRLS_H__
 #define __ATLCTRLS_H__
 
@@ -5067,6 +5070,14 @@
 		return CImageList((HIMAGELIST)::SendMessage(m_hWnd, TB_SETIMAGELIST, 0, (LPARAM)hImageList));
 	}
 
+  // Hydra - begin
+	CImageList SetImageList(HIMAGELIST hImageList, int nImageList)
+	{
+		ATLASSERT(::IsWindow(m_hWnd)); 
+		return CImageList((HIMAGELIST)::SendMessage(m_hWnd, TB_SETIMAGELIST, nImageList, (LPARAM)hImageList));
+	}
+  // Hydra - end	
+
 	CImageList GetDisabledImageList() const
 	{
 		ATLASSERT(::IsWindow(m_hWnd));
